drivers/mfd/mfd_axp192.c

```diff
@@ -3,18 +3,49 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT x_powers_axp192
-
 #include <errno.h>
 #include <stdbool.h>
 
 #include <zephyr/drivers/mfd/axp192.h>
+#include <zephyr/drivers/mfd/axp2101.h>
 #include <zephyr/drivers/i2c.h>
 #include <zephyr/sys/util.h>
 #include <zephyr/logging/log.h>
 
 LOG_MODULE_REGISTER(mfd_axp192, CONFIG_MFD_LOG_LEVEL);
 
+#define AXP_MODEL_COND_11
+#define AXP_MODEL_COND(cond_both_192_2101, cond_192, cond_2101)                                    \
+	COND_CODE_1(UTIL_CAT(UTIL_CAT(AXP_MODEL_COND_,                                             \
+				      CONFIG_DT_HAS_X_POWERS_AXP192_ENABLED),                      \
+			     CONFIG_DT_HAS_X_POWERS_AXP2101_ENABLED),                              \
+		   (cond_both_192_2101),                                                           \
+		   (COND_CODE_1(CONFIG_DT_HAS_X_POWERS_AXP192_ENABLED, (cond_192), (cond_2101))))
+
+#define REG_CHIP_ID(config)                                                                        \
+	AXP_MODEL_COND(config->reg_chip_id, AXP192_REG_CHIP_ID, AXP2101_REG_CHIP_ID)
+#define CHIP_ID(config) AXP_MODEL_COND(config->chip_id, AXP192_CHIP_ID, AXP2101_CHIP_ID)
+#define VBUS_CFG_REG(config)                                                                       \
+	AXP_MODEL_COND(config->vbus_config_reg, AXP192_VBUS_CFG_REG, AXP2101_VBUS_CFG_REG)
+#define VBUS_CFG_VAL_VBUSEN_DISABLE(config)                                                        \
+	AXP_MODEL_COND(config->val_vbusen_disable, AXP192_VBUS_CFG_VAL_VBUSEN_DISABLE,             \
+		       AXP2101_VBUS_CFG_VAL_VBUSEN_DISABLE)
+
+struct mfd_axp192_config {
+	struct i2c_dt_spec i2c;
+#ifdef CONFIG_DT_HAS_X_POWERS_AXP192_ENABLED
+	bool vbusen_disable;
+#endif
+#if CONFIG_DT_HAS_X_POWERS_AXP192_ENABLED && CONFIG_DT_HAS_X_POWERS_AXP2101_ENABLED
+	uint8_t reg_chip_id;
+	uint8_t vbus_config_reg;
+	uint8_t chip_id;
+	uint8_t val_vbusen_disable;
+#endif
+};
```

Comment:
this is confusing and driver unreadable with these changes. just a simple example:

```
#if CONFIG_DT_HAS_X_POWERS_AXP192_ENABLED && CONFIG_DT_HAS_X_POWERS_AXP2101_ENABLED
```

does that mean I now need to have both regulators for the driver to have this section?
Please, refactor to something that makes sense.

Reply:
I did this because I rarely use these two simultaneously, and I don't want to make the table bloated in the typical case (because if there is only one, it can be inlined), but it is a bit roundabout.
If I were to fix it, I would modify it so that this table is always used. Is that OK?

Reply:
@lfelten 

Please apply this patch.
https://github.com/soburi/zephyr/commit/2a0f105f1db074b14f8ea539870704c9254ae7a5

Reply:
Patch applied, thanks.
@soburi do you happen to have a complete datasheet? If so, can you please check register 0x3 / CHIP_ID? It's not documented in my copy.

Reply:
This seems to be a mistake on my part. It seems to no CHIP_ID.

Reply:
https://github.com/hyx0329/axp2101-rs/blob/dev/src%2Fpmu.rs#L561..L567